### DISCUSSION TITLE
Martin table update

### DISCUSSION THREAD

**Discussion by IncidGeo** [maplibre/martin#748] [Q&A] [answered]
Hi everyone,
I run Martin behind nginx in a docker. Our website is base on geodjango/maplibre/j  and it allow the user to feed postgres/postgis via a drawing tool + a turf processing and a form. When a new entity is added with this tool the entity is immediately available in maplibre to be seen.

If i feed postgis with pg_admin or Qgis with a large amount of entity (parcells) in an other table, i need to restart Martin to see the new parcells. How does it exactly work ? Is it a normal behavior ?

**Comment by nyurik** [comment]
Martin would not auto-detect new tables/functions just yet (see https://github.com/maplibre/martin/issues/288), but it should make the new data instantly available without restarts if the data is added/modified in the existing tables/functions.

**Reply by IncidGeo** [reply]
Thank you nyurik,
I've read the #288 post.
It's an existing table. Does Martin need time to update the data ?

**Reply by IncidGeo** [reply]
There is around 1500000 entity for each update

**Reply by nyurik** [reply]
@IncidGeo Martin has no internal caching, at least not at the moment. Martin returns whatever PostgreSQL returns, making a query to PG on every request.

**Reply by IncidGeo** [reply]
Hum perfect for me, but still i don't understand what is happening. I need to restart Martin to see the update .

**Reply by nyurik** [reply]
perhaps browser does some caching?  BTW, Martin doesn't even set the caching/expiry headers (which is not great, we should rethink this - created #749).

**Reply by IncidGeo** [reply]
Nope, I've tried with a "frech" browser wich does'nt even know the website does exist. I've also disable Nginx caching, well i think i have.

```
location / {
        
        proxy_no_cache 1; # No cache
        proxy_cache_bypass 1;# No cache
        proxy_pass http://dgpn_project_server; # Passe a Gunicorn
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;
        proxy_redirect off;
    }

```

**Reply by nyurik** [reply]
Hmm, I am no nginx expert, so won't be much help there. Perhaps try accessing Martin directly to see if the issue remains?

**Reply by IncidGeo** [reply]
I'll find how to do that, thank you Yuri

**Reply by gagecarto** [reply]
@IncidGeo - How did you deal with this situation or did you find a solution? I have the same issue where I need to force Martin reboot to see new data on PostGIS. I am also running behind NGINX.

**Reply by IncidGeo** [reply]
If by 'new data' you mean a new table, I think it's normal behavior. My problem was with a new entity in an existing table. I don't remember exactly what I did, but I think updating to the latest Martin and MapLibre solved the issue.

**Reply by gagecarto** [reply]
For me it's not new tables, but new rows in my tables. Have not yet found a work around and still have to restart Martin to have the new rows return tiles.

**Reply by IncidGeo** [reply, marked as answer]
Sorry, I got confused. I looked at my code and realized it had nothing to do with Nginx or Martin, the issue was on the client side. That said, I still had to update Martin due to a really slow startup problem.

I added the following functions to my code:
```
map.on('zoomend', function () { RefreshZone(); });
map.on('dragend', function () { RefreshZone(); });
```

I also added RefreshZone() at the end of the async functions that adds, saves, and deletes rows in the database.

Then, I defined the RefreshZone function:

```
function RefreshZone() {
    const baseUrl = "{{ base_martin_url }}";
    map.getSource('zones').setUrl(baseUrl + `/carto_projet_zxy`);
    map.getSource('zones_label').setUrl(baseUrl + `/carto_projet_labels_zxy`);
}
```

I guess I could have done better with a script that checks for the latest updates in the table and sets the URL only if needed. This script could also check for updates every, let's say, 5 seconds in case the map is not moved or dragged, and another user has made a change in the table.

**Reply by mlinds** [reply]
@gagecarto I'm having the same issue. New rows in the table are not detected until I restart the Martin container.

**Reply by IncidGeo** [reply]
Have you tried downloading the tile with CURL to make sure the issue isn't coming from the browser cache?

**Reply by CommanderStorm** [reply]
Check if you have set the cache size to something not zero in the config file.
